keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1117 @@
+    egress: 
+      - action: Allow
+        to: 
+        - namespaceSetAndPodSelector:
+          scope: Selector
+          namespaceSelector:
+            matchlabels: 
+              kubernetes.io/metadata.name: kube-system 
+          podSelector: 
+            matchlabels: 
+              app: kube-dns
+```
+
+#### Sample spec for Story 3: Explicitly Delegate traffic to existing K8s Network Policy
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: pub-svc-delegate-example
+spec:
+  priority: 20
+  - appliedTo:
+      namespaceSelector: {}
+    egress:
+    - action: Pass
+      to:
+      - namespaceSetAndPodSelector:
+        scope: Selector 
+        namespaceSelector: 
+          matchLabels:
+            kubernetes.io/metadata.name: bar-ns-1 
+        podSelector: 
+          matchLabels:
+            app: svc-pub
+```
+
+#### Sample spec for Story 4: Create and Isolate multiple tenants in a cluster
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: tenant-creation-example
+spec:
+  priority: 50
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Deny
+       from:
+       - namespaceSetSelector:
+          scope: NotSameLabels
+          labels:
+            - tenant
+```
+
+Note: the above AdminNetworkPolicy can also be written in the following fashion:
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: tenant-creation-example
+spec:
+  priority: 50
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Pass
+       from:
+       - namespaceSetSelector:
+          scope: SameLabels
+          labels:
+            - tenant
+     - action: Deny   # Deny everything else other than same tenant traffic
+       from:
+       - namespaceSetSelector:
+         scope: Selector
+         namespaceSelector: {}
+```
+The difference is that in the first case, traffic within tenant Namespaces will fall
+through, and be evaluated against lower-priority ClusterNetworkPolicies, and then
+NetworkPolicies. In the second case, the matching packet will skip all AdminNetworkPolicy
+evaluation (except for AdminNetworkPolicy priority=baseline), and only match
+against NetworkPolicy rules in the cluster. In other words, the second AdminNetworkPolicy
+specifies intra-tenant traffic must be delegated to the tenant Namespace owners.
+
+#### Sample spec for Story 5: Cluster Wide Default Guardrails
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: baseline-rule-example
+spec:
+  priority: 0
+  - appliedTo:
+     namespaceSelector: {}
+    ingress:
+     - action: Deny   # zero-trust cluster default security posture
+       from:
+       - NamespaceSetSelector
```

Comment:
missing ":" and wrong indent for the next line

Reply:
Fixed